src/nvim/os/stdpaths.c

```diff
@@ -69,15 +69,16 @@ const char *get_appname(void)
   return env_val;
 }
 
-/// Ensure that APPNAME is valid. In particular, it cannot contain directory separators.
+// Ensure that APPNAME is valid. Must be a name or relative path.
 bool appname_is_valid(void)
 {
   const char *appname = get_appname();
-  const size_t appname_len = strlen(appname);
-  for (size_t i = 0; i < appname_len; i++) {
-    if (appname[i] == PATHSEP) {
-      return false;
-    }
+  if (path_is_absolute(appname)
+      || strcmp(appname, ".") == 0
+      || strcmp(appname, "..") == 0
+      || strstr(appname, PATHSEPSTR"..") != NULL
+      || strstr(appname, ".."PATHSEPSTR) != NULL) {
```

Comment:
unfortunately this fails on windows bc of the way PATHSEPSTR is defined: https://github.com/neovim/neovim/blob/5ea5812714bcbeb6a5438c80d410984ad4fbb922/src/nvim/ascii.h#L81

let's just do this for now (both "/" and "\\" need to be checked on Windows, unless we "normalize" appname before this code is reached):

```suggestion
#ifdef BACKSLASH_IN_FILENAME
      || strstr(appname, "\\..") != NULL
      || strstr(appname, "..\\") != NULL)
#endif
      || strstr(appname, "/..") != NULL
      || strstr(appname, "/..") != NULL) {
```